datadog_lambda/wrapper.py

```diff
@@ -279,8 +278,6 @@ def _before(self, event, context):
             self.response = None
             set_cold_start(init_timestamp_ns)
             submit_invocations_metric(context)
-            if is_legacy_lambda_step_function(event):
-                event = event["Payload"]
```

Comment:
Moved this unwrapping to happen inside of `tracing.extract_context_from_step_functions()`